### ISSUE TITLE
[no-var-requires] false error on *.js files

### ISSUE THREAD

**Issue by guoyunhe** [typescript-eslint/typescript-eslint#1724]
**Repro**

The full code repo is here: https://github.com/advclb/react-button

```JSON
{
  "env": {
    "node": true,
    "browser": true,
    "es6": true
  },
  "extends": [
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:prettier/recommended",
    "plugin:import/typescript",
    "prettier/@typescript-eslint",
    "prettier/react"
  ],
  "globals": {
    "Atomics": "readonly",
    "SharedArrayBuffer": "readonly"
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": 2018,
    "sourceType": "module"
  },
  "plugins": ["react", "prettier"],
  "rules": {
    "react/prop-types": "off"
  },
  "settings": {
    "import/parsers": {
      "@typescript-eslint/parser": [".ts", ".tsx"]
    },
    "import/extensions": [".js", ".jsx", ".ts", ".tsx"]
  }
}
```

I have a webpack.config.js file:

```js
const path = require("path");
const webpack = require("webpack");
const { CleanWebpackPlugin } = require("clean-webpack-plugin");
```

**Expected Result**

ESLint shouldn't report errors on `var = require()` in JS files because they are not typescript.

**Actual Result**

ESLint reports errors on `var = require()` in JS file.

```
/home/xxx/git/advclb/react-button/webpack.config.js
  1:14  error  Require statement not part of import statement  @typescript-eslint/no-var-requires
  2:17  error  Require statement not part of import statement  @typescript-eslint/no-var-requires
  3:32  error  Require statement not part of import statement  @typescript-eslint/no-var-requires
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.23.0` |
| `@typescript-eslint/parser`        | `2.22.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.14.2` |

**Comment by bradzacher** [comment]
Our rules intentionally do not distinguish between JS and TS files (for a number of reasons).

You can use eslint's overrides feature to disable incompatible rules on JS files.
https://eslint.org/docs/user-guide/configuring#disabling-rules-only-for-a-group-of-files
